configs/vue3.js

```diff
@@ -1,6 +1,6 @@
 module.exports = {
   extends: [
-    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-type-checked',
```

Comment:
I **think** we might also need `'plugin:@typescript-eslint/stylistic-type-checked',` but I find https://typescript-eslint.io/blog/announcing-typescript-eslint-v6/ to be very confusing

Reply:
I just went through the docs and indeed, it makes sense to add it